src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +464,125 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
         return this;
     }
 
+    /**
+     * Use {@link #withFIPSEnabled(Path)} with default value of {@code target/dependency}
+     * @return
+     */
+    public RealJenkinsRule withFIPSEnabled() {
+        return withFIPSEnabled(Path.of("target", "dependency"));
+    }
+
+    /**
+     * Enable FIPS mode assuming all FIPS BouncyCastle jars (bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar) are in the provided directory.
+     * This will set some System properties such :
+     * <ul>
+     *   <li>-Xbootclasspath with FIPS BouncyCastle jars</li>
+     *   <li>java.security.properties to a new temp file configuring the security provider to FIPS BouncyCastle</li>
+     *   <li>-Dsecurity.overridePropertiesFile=true</li>
+     *   <li>-Djavax.net.ssl.trustStoreType=PKCS12</li>
+     *   <li>-Djenkins.security.FIPS140.COMPLIANCE=true</li>
+     *   <li>-Dcom.redhat.fips=false</li>
+     * </ul>
+     * @param fipsLibrariesPath path to directory with FIPS BouncyCastle jars
+     * @return
+     */
+    public RealJenkinsRule withFIPSEnabled(@NonNull Path fipsLibrariesPath) {
+        this.fipsLibrariesPath = Objects.requireNonNull(fipsLibrariesPath);
+        // check expected BouncyCastle fips jars are here
+        if  (!Files.exists(this.fipsLibrariesPath) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bc-fips.jar")) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bctls-fips.jar")) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bcpkix-fips.jar"))) {
+            throw fipsIllegalSetupException(this.fipsLibrariesPath);
+        }
+
+        withBootClasspath(fipsLibrariesPath.resolve("bc-fips.jar").toFile(),
+                            fipsLibrariesPath.resolve("bctls-fips.jar").toFile(),
+                            fipsLibrariesPath.resolve("bcpkix-fips.jar").toFile());
+        try {
+            javaOptions("--add-exports", "java.base/sun.security.provider=ALL-UNNAMED",
+                            "-Dsecurity.overridePropertiesFile=true",
+                            "-Djava.security.properties=" + writeFIPSJavaSecurityFile().toUri(),
+                            "-Dorg.bouncycastle.fips.approved_only=true",
+                            "-Djavax.net.ssl.trustStoreType=PKCS12",
+                            "-Djenkins.security.FIPS140.COMPLIANCE=true",
+                            "-Dcom.redhat.fips=false");
```

Comment:
if you are disabling the RHEL patches, then you should not need to adjust `fips.provider.*` (or the add-exports option).

Reply:
fixed